### ISSUE TITLE
leap webapp tests fail on node with plain couchdb

### ISSUE THREAD

**Issue by varac** [pixelated-project/pixelated-platform#101]
i'm using the same couchdb 1.6 setup as described in https://github.com/pixelated-project/pixelated-platform/issues/91#issuecomment-114073944

```
--- pixelated-platform-environments/pixelated-project.org ‹couch16* M› » leap test --continue 
 = [couch16.local.pixelated-project.org] PASS: Network > Can connect to internet?
 = [couch16.local.pixelated-project.org] PASS: Network > Is stunnel running?
 = [couch16.local.pixelated-project.org] PASS: Network > Is shorewall running?
 = [couch16.local.pixelated-project.org] PASS: CouchDB > Are daemons running?
 = [couch16.local.pixelated-project.org] PASS: CouchDB > Is CouchDB running?
 = [couch16.local.pixelated-project.org] PASS: CouchDB > Do ACL users exist?
 = [couch16.local.pixelated-project.org] PASS: CouchDB > Do required databases exist?
 = [couch16.local.pixelated-project.org] PASS: CouchDB > Can records be created?
 = [couch16.local.pixelated-project.org] PASS: Mx > Can contact couchdb?
 = [couch16.local.pixelated-project.org] PASS: Mx > Can contact couchdb via haproxy?
 = [couch16.local.pixelated-project.org] PASS: Mx > Are MX daemons running?
 = [couch16.local.pixelated-project.org] PASS: Soledad > Is Soledad running?
 = [couch16.local.pixelated-project.org] PASS: Webapp > Can contact couchdb?
 = [couch16.local.pixelated-project.org] PASS: Webapp > Can contact couchdb via haproxy?
 = [couch16.local.pixelated-project.org] PASS: Webapp > Are daemons running?
 = [couch16.local.pixelated-project.org] PASS: Webapp > Can access webapp?
 = [couch16.local.pixelated-project.org] FAIL: Webapp > Can create and authenticate and delete user via API? [webapp.rb:46]:
 = [couch16.local.pixelated-project.org]   Unable to check that user identity was deleted: HTTP response from API should have code 200, was 401  {"error":"unauthorized","reason":"You are not authorized to access this db."}
 = [couch16.local.pixelated-project.org] FAIL: Webapp > Can sync Soledad? [webapp.rb:59]:
 = [couch16.local.pixelated-project.org]   Unable to check that user identity was deleted: HTTP response from API should have code 200, was 401  {"error":"unauthorized","reason":"You are not authorized to access this db."}
 = [couch16.local.pixelated-project.org] 20 tests: 16 passes, 0 skips, 0 warnings, 2 failures, 0 errors

```

**Comment by bwgpro** [comment]
hey, I am blocked on this because of the same problem as #91. I'll add more info there

**Comment by bwgpro** [comment]
Fixed one of the issues, the webapp can now delete the test user again.
The problem now is that tapicero is lost and can't create user dbs, it corrupt the user sequence too to make sure it cannot recover never ever. I'll have to take a better look at the tapicero problem.

**Comment by varac** [comment]
@bwagnerr could you start a new issue for the tapicero problem so we don't mix things here pls ?
and where/how did you fix this issue ?

**Comment by bwagnerr** [comment]
Hey @varac sorry for the delay, I pushed the fix to leap platform. I was trying to take a look at the tapicero problem again but got blocked by the NTP, I can't recreate my environment here.

I tried connecting through a VPN, but the guest won't use it, I was taking a look at the vagrantfile and it seems like there's already a workaround that was supposed to fix that, but I still stumble on the same error. Do you have any ideas how to fix this? If the guest would use the VPN the host is using I would probably be able to bypass this problem for now.

**Comment by bwagnerr** [comment]
I fixed the tests but tapicero is broken with couchdb 1.6.
I created the issue #102 to investigate that, this issue can be tested but the tests that deal with tapicero will break and should be looked upon on the other issue

**Comment by fbernitt** [comment]
These are the results of the test run:
```
...
 = [couch16.local.pixelated-project.org] PASS: CouchDB > Are daemons running?
 = [couch16.local.pixelated-project.org] PASS: CouchDB > Is CouchDB running?
 = [couch16.local.pixelated-project.org] PASS: CouchDB > Do ACL users exist?
 = [couch16.local.pixelated-project.org] PASS: CouchDB > Do required databases exist?
 = [couch16.local.pixelated-project.org] PASS: CouchDB > Can records be created?
 = [couch16.local.pixelated-project.org] PASS: Mx > Can contact couchdb?
 = [couch16.local.pixelated-project.org] PASS: Mx > Can contact couchdb via haproxy?
 = [couch16.local.pixelated-project.org] FAIL: Mx > Are MX daemons running? [mx.rb:35]:
 = [couch16.local.pixelated-project.org]   No running process for leap_mx
 = [couch16.local.pixelated-project.org] FAIL: Webapp > Can sync Soledad? [webapp.rb:59]:
 = [couch16.local.pixelated-project.org]   Could not find user db `user-2bb6c02af9901d53595359b6101e63f8` for test user `test_user_2hbe8rasus4cua`
 = [couch16.local.pixelated-project.org]   uuid=2bb6c02af9901d53595359b6101e63f8
 = [couch16.local.pixelated-project.org]   HTTP 404  {"error":"not_found","reason":"no_db_file"}

 = [couch16.local.pixelated-project.org] 20 tests: 16 passes, 0 skips, 0 warnings, 2 failures, 0 errors
```

**Comment by bwagnerr** [comment]
FAIL: Webapp > Can sync Soledad?
This happens because tapicero is broken with couch 1.6, that's what issue #102 is for
FAIL: Mx > Are MX daemons running?
This one didn't happen for me, I just double checked and it doesn't happen here, do you have any logs for leap_mx?

**Comment by fbernitt** [comment]
There is nothing special in the logs. The service is stopped and then never restarted. if I start it manually afterwards and then run the tests , only the 'Can sync Soledad?' one fails.

```
015-07-27 07:28:57+0000 [-] Log opened.
2015-07-27 07:28:57+0000 [-] twistd 13.2.0 (/usr/bin/python 2.7.3) starting up.
2015-07-27 07:28:57+0000 [-] reactor class: twisted.internet.epollreactor.EPollReactor.
2015-07-27 07:28:57+0000 [-] AliasResolverFactory starting on 4242
2015-07-27 07:28:57+0000 [-] Starting factory <leap.mx.alias_resolver.AliasResolverFactory object at 0x2518a90>
2015-07-27 07:28:57+0000 [-] CheckRecipientAccessFactory starting on 2244
2015-07-27 07:28:57+0000 [-] Starting factory <leap.mx.check_recipient_access.CheckRecipientAccessFactory object at 0x2518ad0>
2015-07-27 07:28:57+0000 [-] Failed adding watch to FilePath('/var/mail/vmail/Maildir'), will try again in 300 seconds: Not a directory: '/var/mail/vmail/Maildir'
2015-07-27 07:28:57+0000 [-] Starting processing skipped mail...
2015-07-27 07:28:57+0000 [-] --------------------------------------------------
2015-07-27 07:28:57+0000 [-] ++++++++++++++++++++++++++++++++++++++++++++++++++
2015-07-27 07:28:57+0000 [-] Done processing skipped mail
2015-07-27 07:29:05+0000 [-] Received SIGTERM, shutting down.
2015-07-27 07:29:05+0000 [-] (TCP Port 2244 Closed)
2015-07-27 07:29:05+0000 [-] Stopping factory <leap.mx.check_recipient_access.CheckRecipientAccessFactory object at 0x2518ad0>
2015-07-27 07:29:05+0000 [-] (TCP Port 4242 Closed)
2015-07-27 07:29:05+0000 [-] Stopping factory <leap.mx.alias_resolver.AliasResolverFactory object at 0x2518a90>
2015-07-27 07:29:05+0000 [-] Main loop terminated.
2015-07-27 07:29:05+0000 [-] Server Shut Down.
```
As far as this ticket is concerned, I think we can close it.

**Comment by varac** [comment]
i'd like to re-open this again, as stated in the gdoc: "it's blocking an important issue for our goal". I'd like to stabilize the platform side of things and couch 1.6 would possibly help here a lot over using bigcouch (which is based on couch 1.1).

**Comment by bwagnerr** [comment]
hey @varac, maybe we should have an issue (here or leap) to make tapicero compatible with couchdb 1.6, that is the only remaining issue, I don't think we should deal with that on this one

**Comment by varac** [comment]
sorry, i meant to propose to reopen #102 which tracks the tapicero bug with couch 1.6
